test_scripts/Defects/8_1/3838_2_SDL_does_not_send_UI.ClosePopUp_in_case_VR.PerformInteraction_SUCCESS.lua

```diff
@@ -0,0 +1,178 @@
+  vrHelp = vrHelpvalue,
+  interactionLayout = "ICON_ONLY"
+}
+
+--[[ Local Functions ]]
+local function setChoiceSet(pChoiceIDValue)
+  local temp = {
+    {
+      choiceID = pChoiceIDValue,
+      menuName = "Choice" .. tostring(pChoiceIDValue),
+      vrCommands = {
+        "VrChoice" .. tostring(pChoiceIDValue),
+      }
+    }
+  }
+  return temp
+end
+
+local function sendOnSystemContext(pCtx)
+  common.getHMIConnection():SendNotification("UI.OnSystemContext", {
+    appID = common.getHMIAppId(),
+    systemContext = pCtx
+  })
+end
+
+local function expectOnHMIStatusWithAudioStateChanged_PI(pRequest)
+  if pRequest == "BOTH" then
+    common.getMobileSession():ExpectNotification("OnHMIStatus",
+      { hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" },
+      { hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE", systemContext = "VRSESSION" },
+      { hmiLevel = "FULL", audioStreamingState = "ATTENUATED", systemContext = "VRSESSION" },
+      { hmiLevel = "FULL", audioStreamingState = "ATTENUATED", systemContext = "HMI_OBSCURED" },
+      { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "HMI_OBSCURED" },
+      { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN" })
+    :Times(6)
+  elseif pRequest == "VR" then
+    common.getMobileSession():ExpectNotification("OnHMIStatus",
+      { systemContext = "MAIN", hmiLevel = "FULL", audioStreamingState = "ATTENUATED" },
+      { systemContext = "MAIN", hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE" },
+      { systemContext = "VRSESSION", hmiLevel = "FULL", audioStreamingState = "NOT_AUDIBLE" },
+      { systemContext = "VRSESSION", hmiLevel = "FULL", audioStreamingState = "AUDIBLE" },
+      { systemContext = "MAIN", hmiLevel = "FULL", audioStreamingState = "AUDIBLE" })
+    :Times(5)
+  elseif pRequest == "MANUAL" then
+    common.getMobileSession():ExpectNotification("OnHMIStatus",
+      { systemContext = "MAIN", hmiLevel = "FULL", audioStreamingState = "ATTENUATED" },
+      { systemContext = "HMI_OBSCURED", hmiLevel = "FULL", audioStreamingState = "ATTENUATED" },
+      { systemContext = "HMI_OBSCURED", hmiLevel = "FULL", audioStreamingState = "AUDIBLE" },
+      { systemContext = "MAIN", hmiLevel = "FULL", audioStreamingState = "AUDIBLE" })
+    :Times(4)
+  end
+end
+
+local function createInteractionChoiceSet(pChoiceSetID)
+  local cid = common.getMobileSession():SendRPC("CreateInteractionChoiceSet", {
+      interactionChoiceSetID = pChoiceSetID,
+      choiceSet = setChoiceSet(pChoiceSetID),
+    })
+  common.getHMIConnection():ExpectRequest("VR.AddCommand", {
+      cmdID = pChoiceSetID,
+      type = "Choice",
+      vrCommands = { "VrChoice" .. tostring(pChoiceSetID) }
+    })
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+  common.getMobileSession():ExpectResponse(cid, { resultCode = "SUCCESS", success = true })
+end
```

Comment:
Looks like most of these functions are copied between tests, they should be moved out into a common file